db/db_impl_write.cc

```diff
@@ -341,11 +344,22 @@ Status DBImpl::PreprocessWrite(const WriteOptions& write_options,
   }
 
   if (status.ok() && need_log_sync) {
+    // Wait until the parallel syncs are finished. Any sync process has to sync
+    // the front log too so it is enough to check the status of front()
+    // We do a while loop since log_sync_cv_ is signalled when any sync is finished
+    // Note: there does not seem to be a reason to wait for parallel sync at
```

Comment:
I don't remember much, but this seems accurate. Maybe the waiting can be solved by turning getting_synced into a counter instead of bool, but as you said SyncWAL and need_log_sync are usually not used together.